### PR TITLE
Task/angular rc5 forms

### PR DESCRIPTION
**Pull request by jgf20**
- Update forms to RC5 in all carbonldp-panel
    - login, create-app, edit-app, backup-importer, property, pointer, literal components.
- Add new validators in custom-validators
- Add directives.module to hold customize validators and directives.

### CODE REVIEW

File: src/document-explorer/literals/literal.component.html [hunk 1]
```diff
@@ -2,13 +2,15 @@
 	<div class="read-mode" *ngIf="mode===modes.READ">
 		<p class="value">{{value}}</p>
 	</div>
+	<!--<div class="edit-mode ui error form">-->
 	<div class="edit-mode ui error form" *ngIf="mode===modes.EDIT">
 		<div class="field" [ngClass]="{error:!valueInput.valid}">
-			<input type="text" [ngFormControl]="valueInput" (ngModelChange)="value=$event">
+			<input type="text" name="valueInput" [(ngModel)]="value" #valueInput="ngModel" [type]="type" [control]="valueInput" required value-validator>
```
Comment (MiguelAraCo) [inline src/document-explorer/literals/literal.component.html]: HTML attributes must be semantic. "value-validator" is not an attribute of the input. Instead the input should be tagged with an attribute like: "rdfValue". Something that states the nature of the input.

File: src/document-explorer/property/property.component.ts [hunk 2]
```diff
@@ -104,17 +108,22 @@ export class PropertyComponent implements AfterViewInit, OnInit {
...
+		// if( this.idInputControl )this.idInputControl.control.updateValueAndValidity(this.value);
+		// if( this.nameInputControl )this.nameInputControl.control.updateValueAndValidity(this.name);
+		// console.log(this.nameInput, this.idInput);
 	}
 
 	getDisplayName( uri:string ):string {
 		if( this.commonToken.indexOf( uri ) > - 1 )return uri;
 		if( URI.Util.hasFragment( uri ) )return this.unescape( this.getFragment( uri ) );
 		return this.unescape( URI.Util.getSlug( uri ) );
+		// return ""; //todo remove this line
```
Comment (MiguelAraCo) [inline src/document-explorer/property/property.component.ts]: TODOs are written following this syntax: `// TODO: Remove this line`. What's preventing this line from being removed?

File: src/document-explorer/property/property.component.ts [hunk 3]
```diff
@@ -143,7 +155,7 @@ export class PropertyComponent implements AfterViewInit, OnInit {
 	}
 
 	getTypeIcon( type:string ):string {
-		switch( this.getDisplayName( type ) ) {
+		switch ( this.getDisplayName( type ) ) {
```
Comment (MiguelAraCo) [inline src/document-explorer/property/property.component.ts]: Remember to format all source code before creating a pull request.

**Comment by MiguelAraCo** [comment]
All `json` files were formatted. Please be careful with those files because they'll be automatically modified by `npm` or `jspm`. They should follow their natural format so they don't appear on each pull request.

**Comment by codyburleson** [comment]
Great tip on modifying keymap to do formatting when pressing CMD + SHIFT + S instead of executing it on each CMD + S!!!

**Pull request merged**